install_sct

```diff
@@ -165,14 +171,18 @@ function check_requirements() {
     print warning "WARNING: \"gcc\" is not installed."
     if [[ $OS == "osx" ]]; then
       while [[ ! $GCC_INSTALL =~ ^([Yy](es)?|[Nn]o?)$ ]]; do
-        print question "Do you want to install it now? (accepting to install \"gcc\" will also install \"brew\" in case it is not installed already)? [y]es/[n]o: "
-        read GCC_INSTALL
+        GCC_INSTALL=yes # rude?
+        if [ -z "$NONINTERACTIVE" ]; then
+          print question "Do you want to install it now? (accepting to install \"gcc\" will also install \"brew\" in case it is not installed already)? [y]es/[n]o: "
+          read GCC_INSTALL
+        fi
```

Comment:
I think I agree with the `# rude?` note here. I think I want to keep this prompt. The intent of `-y` is usually just to silence the 3 most common questions. Any issues with `gcc` would be a rare occurrence, so the interruption won't be too bothersome, even with `-y`. 

Plus, installing `gcc`+`homebrew` is a bit heavier, so I think the user should have the option to decline regardless of `-y`.

Reply:
I meant `-y` to run the whole thing interactively, which means either the script should complete successfully or it should error; it shouldn't hang in the middle.

So in that case I would set the default to `no` I think. In the rare case that someone is installing non-interactively (for CI or deploying to a computing cluster or something) and also doesn't have gcc installed already I think we should just let it fail. We should give it a shot to see what that looks like on a fresh macOS VM; maybe we can print a warning like "Warning: gcc is not installed. The installation might fail." 

Julien's attempt last night to actually use this in #3108 makes me think it's -very very rare-.

Reply:
I really think this entire code block should go away so I don't want to maintain it too hard.

We should be using better packaging tools instead of trying to futz around with this install script. We can package SCT *in* brew itself. If we're going to potentially make people install brew anyway let's just do that.

Reply:
> I meant -y to run the whole thing interactively, which means either the script should complete successfully or it should error; it shouldn't hang in the middle.

Ah, that makes a lot more sense. My usage was similar to Julien's... sometimes I am too lazy to want to answer all of the questions when I just want the whole thing to run in the background. :sweat_smile: 

> maybe we can print a warning like "Warning: gcc is not installed. The installation might fail."

I think this is a good idea! Good thing we already have that warning. :stuck_out_tongue_winking_eye: 

https://github.com/neuropoly/spinalcordtoolbox/blob/3761f16569bf44df819e39127665e7dab2c5a432/install_sct#L171

Reply:
I changed the default to "no"!